src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,328 @@
+    # Load Grant requests
+    grant_record_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_record_2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_1.json')))
+    grant_record_3 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_2.json')))
+    grant_record_4 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_3.json')))
+    
+    #Load CBSD
+    device_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    device_3 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    device_4 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_d.json')))
+
+    # Load GWPZ Record
+    gwpz_record_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'gwpz_record_0.json')))
+
+    # Load FSS record
+    fss_record_1 =  json.load(
+        open(os.path.join('testcases', 'testdata', 'fss_record_0.json')))
+    fss_record_2 =  json.load(
+        open(os.path.join('testcases', 'testdata', 'fss_record_1.json')))
+
+    # Load ESC record
+    esc_record_1 =  json.load(
+        open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json')))
+
+
+    # Create the actual config.
+    config = {
+        'cbsdRecords': [{'registrationRequests':[device_1,device_2],'grantRequests':[grant_record_1,grant_record_2]},
+                        {'registrationRequests':[device_3,device_4],'grantRequests':[grant_record_3,grant_record_4]}],
+        'IAPRecords': [[
+      { 'type': 'FSS','Record': [fss_record_1] },
+      { 'type': 'ESC','Record': [esc_record_1] } ],
+      [ { 'type': 'GWPZ','Record': [gwpz_record_1]},
+      { 'type': 'FSS', 'Record': [fss_record_2]}]],
+	'sas_hareness': [{'name': 'SAS-TH-1',
+			'url': ['localhost', '9001'],
+			'cert_file': 'certs/server_a.cert',
+			'key_file': 'certs/server_a.key'},
+	                {
+			'name': 'SAS-TH-2',
+			'url': ['localhost', '9002'],
+			'cert_file': 'certs/server_b.cert',
+			'key_file': 'certs/server_b.key'
+		        }],
+        'sas_uut': {'name': 'SAS-UUT',
+                        'url': ['localhost', '9000'],
+                        'cert_file': 'certs/server.cert',
+                        'key_file': 'certs/server.key'}, 
+        'K':3, 
+        'headroom':{'MgPpa':2,'MgGwpz':2,'MgCochannel':2,'MgBlocking':2,'MgOobe':2,'MgEsc':2},
+        'DPAactiveInfo':[],
+        'DPAdeactiveInfo':[]
+       }
+    
+    writeConfig(filename, config)
+
+
+  @configurable_testcase(generate_MCP_1_default_config) 
+  def test_WINNF_FT_S_MCP_1(self,config_filename):
+    """SAS manages a mix of GAA and PAL Grants in 3550 MHz
+       to 3700 MHz to protect configurable IAP-protected entities and DPAs
+    """
+    config = loadConfig(config_filename)
+    pre_iap_headroom = config ['headroom']
```

Comment:
Let's have this information stored in a file similar to sas.cfg -- the headroom value will be common to all SASes and does not need to be configurable within the test case.

Reply:
we will   check if any other MCP related test cases have dependency? For all the MCP   related TCs we will add a new configuration file